crates/bevy_animation/src/lib.rs

```diff
@@ -124,10 +126,59 @@ impl Default for PlayingAnimation {
             elapsed: 0.0,
             animation_clip: Default::default(),
             path_cache: Vec::new(),
+            root_motion: None,
         }
     }
 }
 
+#[derive(Reflect, FromReflect, Default, Clone)]
+enum RootMotionMode {
+    #[default]
+    Enabled,
+    Locked,
+}
+
+/// Controls how root motion is handled during an animation
+#[derive(Reflect, Default, FromReflect, Clone)]
+pub struct RootMotion {
+    mode: RootMotionMode,
+    root_node: EntityPath,
+    root_transform_y: bool,
+    root_transform_xz: bool,
+    previous_translation: Vec3,
+}
+
+impl RootMotion {
+    /// Take the root motion for the specified bone projected on the ground plane and transfer it to the parent entity
+    pub fn on_bone(path: EntityPath) -> RootMotion {
+        RootMotion {
+            mode: RootMotionMode::Enabled,
+            root_node: path,
+            root_transform_y: false,
+            root_transform_xz: true,
+            previous_translation: Vec3::ZERO,
+        }
+    }
+
+    /// Do not transfer root motion to the parent entity
+    pub fn locked(mut self) -> Self {
```

Comment:
Personally, I think these methods should be reversible and that there should be a way to modify the `RootMotion` from `AnimationPlayer`, not just overriding it. Maybe keep these methods for the sake of the builder pattern, but make the fields public and add `root_motion` and `root_motion_mut` methods to `AnimationPlayer`?

Reply:
These comments could also be transferred to the fields, if they become public. They also need proper punctuation to fit doc comment standards.